### ISSUE TITLE
Shiny application breaks when logging in, not sure why. 

### ISSUE THREAD

**Issue by Saimon-Khan** [ministryofjustice/data-platform-support#311]
### Which tool do you need help with?

RStudio

### What happened?

Hello,

I've tried to deploy the dev version of my application. I have tested the app locally which was fine and the workflow on GitHub ran successfully too, I am not sure what's going wrong. I've checked the logs on Kibana, but I can't seem to understand the issue. Could someone help please? 

This is my app: https://github.com/ministryofjustice/cjsm-monitoring/actions
This is the link to the app itself: https://cjsm-monitoring-dev.apps.live.cloud-platform.service.justice.gov.uk/

**Comment by jhpyke** [comment]
Hi,

Looking through Kibana, we can see an error:

```{"level":"info","message":"Warning: Error in loadNamespace: there is no package called ‘markdown’\n","service":"shiny.stderr"}```

If you install `markdown` specifically during the creation of your image and then deploy the new version, can you see if this resolves the deployment issue?

Thanks,
Jake

**Comment by Saimon-Khan** [comment]
Hi Jake, 

Thanks for your response, finally seeing some light! 
I checked my renv.lock file and noticed there's no markdown. I tried installing it through renv, but it still does not appear there, really not sure what's going wrong. 
On the other hand, there's a problematic python requirement called `pkg_resources==0.0.0` inside my `requirements.txt` file. Whenever I tried to run the workflow on GitHub, it would error out with this being the issue. Every time I snasphot with renv, this package comes back...do you have any idea how to fix that as well?

**Comment by jhpyke** [comment]
To clarify on the 'creation of your image':

[This file](https://github.com/ministryofjustice/cjsm-monitoring/blob/main/Dockerfile) is used to control what setup processes your app goes through when deploying. In this case, we can see it runs:

```
RUN R -e "install.packages('renv')"
RUN R -e 'renv::restore()'

ADD requirements.txt requirements.txt
RUN pip3 install -r requirements.txt
```

To install your renv.lock, and any requirements in requirements.txt.

On the issue of `pkg_resources==0.0.0` - this is a [known issue](https://bugs.launchpad.net/ubuntu/+source/python-pip/+bug/1635463) of how Linux interacts with certain versions of Python/Pip. You could likely fix it with a newer version of python/pip in your local environment where you're snapshotting, but this may require other work due to changed package versions. If you're simply looking to release an image, you should be safe to remove the line manually before committing the new `requirements.txt`

**Comment by Saimon-Khan** [comment]
Thanks Jake, I've added a line to the docker file in my dev branch. 
I find it weird that I am unable to write the markdown package to the renv.lock there. 

Regardless, after doing that and running a new workflow, it still doesn't seem to load. I've checked the logs in the build and it doesn't seem to capture the change I've made to the docker. 

I noticed that when I run the workflow, the "Deploy helm chart into Cloud platform" doesn't run anymore. Do you know if this is the issue? Am I doing something wrong?

**Comment by jhpyke** [comment]
So the deployment of a new chart only triggers [on a Pull Request](https://github.com/ministryofjustice/cjsm-monitoring/actions/runs/6704982175/workflow#L81-L84). As such, any manual triggers of the workflow will only cause the image to be built, but not to be deployed. This is intended to allow you to rapidly debug build errors, but I understand that this behaviour isn't the clearest without going through the file. As such, if you make any changes via a PR, you should be able to deploy the new version.

**Comment by Saimon-Khan** [comment]
Hi Jake,
Thanks a lot for your help, I was able to successfully deploy it this time around through a PR. I didn't know it had to be done through a pull request specifically. 

Regarding the markdown issue, it seems to be a recent one. [https://github.com/rstudio/renv/issues/1708](url)
renv doesn't add markdown to the lockfile anymore which is what was giving issues during the deployment, but I have fixed it by basically removing the markdown part of the code (in my case, I didn't have to specifically use it so I just migrated the code)

Thanks,
Saimon
